clang/test/CodeGenSYCL/device_global_int_footer_header.cpp

```diff
@@ -52,8 +52,13 @@ int main() {
 // CHECK-FOOTER-NEXT: namespace {
 // CHECK-FOOTER-NEXT: __sycl_device_global_registration::__sycl_device_global_registration() noexcept {
 
+extern device_global<int> Basic;
 device_global<int> Basic;
 // CHECK-FOOTER-NEXT: device_global_map::add((void *)&::Basic, "_Z5Basic");
+// CHECK-FOOTER-NOT: Basic
```

Comment:
Thanks for adding that test.   Quick question:  what does this NOT line check for?

Reply:
My understanding was that it is to check that the variable `Basic` is not added twice because of external and normal declaration.

Reply:
Yes. We want to add only the definition. Not the declaration.